src/arithmetic/arithmetic_expression.c

```diff
@@ -87,6 +90,11 @@ static AR_ExpNode *_AR_EXP_CloneOp(AR_ExpNode *exp) {
 	if(exp->op.type == AR_OP_FUNC) {
 		clone->op.f = exp->op.f;
 		clone->op.type = AR_OP_FUNC;
+		if(exp->op.f->bclone) {
+			// If a function has its own clone routine, it will handle all of its children.
+			exp->op.f->bclone(exp, clone);
```

Comment:
I think I've missed understood the purpose of the `bclone` function, why the current AR clone logic isn't sufficient?